src/thing_creature.c

```diff
@@ -6371,12 +6372,20 @@ void controlled_creature_drop_thing(struct Thing *creatng, struct Thing *droptng
                         if (creature_is_being_unconscious(droptng))
                         {
                             struct CreatureControl* dropctrl = creature_control_get_from_thing(droptng);
+                            //creature already have a lair rom
                             if (dropctrl->lair_room_id == room->index)
                             {
                                 make_creature_conscious(droptng);
                                 initialise_thing_state(droptng, CrSt_CreatureGoingHomeToSleep);
-                                dropctrl->flgfield_1 |= CCFlg_NoCompControl;
                             }
+                            //creature doesn't have a lair room but she will and can sleep here
+                            if(game.conf.rules.workers.drag_to_lair == 2 && dropctrl->lair_room_id == 0 && creature_can_do_healing_sleep(droptng) && room_has_enough_free_capacity_for_creature_job(room, droptng, Job_TAKE_SLEEP))
+                            {
+                                make_creature_conscious(droptng);
+                                initialise_thing_state(droptng, CrSt_CreatureChangeLair);
+                            }
+                            dropctrl->flgfield_1 |= CCFlg_NoCompControl;
```

Comment:
We have the `set_flag` function for this now:
`set_flag(dropctrl->flgfield_1,CCFlg_NoCompControl);`